firmware/application/apps/ui_mictx.cpp

```diff
@@ -214,16 +223,22 @@ MicTXView::MicTXView(
 	};
 	options_gain.set_selected_index(1);		// x1.0
 	
+	tx_frequency = transmitter_model.tuning_frequency();
 	field_frequency.set_value(transmitter_model.tuning_frequency());
 	field_frequency.set_step(receiver_model.frequency_step());
 	field_frequency.on_change = [this](rf::Frequency f) {
-		this->on_tuning_frequency_changed(f);
+		tx_frequency = f;
+		if(!rx_enabled)
```

Comment:
Suggestion: This "if" is not needed, since you may inject the frequency at the set_tx() function, when starting transmission, just like with tx_gain and rf_amp